.github/workflows/tag-version.yml

```diff
@@ -0,0 +1,62 @@
+# Creates a tag and a github release for new version
+
+name: Tag the commit
+
+on:
+  workflow_call:
+    inputs:
+      branch:
+        description: "Default branch of the target repository"
+        required: false
+        default: "master"
+        type: string
+
+jobs:
+  tag_release:
+    if: ${{ github.event.pull_request.merged && contains(github.head_ref,'-update-changelogs')}}
+    runs-on: ubuntu-20.04
+    outputs:
+      output1: ${{ steps.gitversion.outputs.version }}
+      output2: ${{ steps.changedfiles.outputs.files }}
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        token: ${{ secrets.PAT_TAG_PUSH }}
```

Comment:
We'll have to add a new secret in our organisation to trigger pypi-publish workflow when tag is pushed via this tag-version workflow. [Reference](https://github.community/t/push-from-workflow-not-doesnt-trigger-on-push-tags-workflow/149151)